### ISSUE TITLE
Error 503 on new user login

### ISSUE THREAD

**Issue by evanp** [producthunt/producthunt-api#58]
I had a new user log in with Product Hunt on my app https://personalhunt.com/ . They had an error
![undefined_-_personal_hunt](https://cloud.githubusercontent.com/assets/6770/7416370/cdb64a48-ef2c-11e4-8816-22c95a4b503f.png) returning a 503 status code, but just showing the "500.html" content. This is when converting the authorization token to an access token.

The screenshot just shows my default dev-mode error reporting with the HTML body of the API error.

It was the first invited member who'd logged in, so I thought it might be that, but then I mooked an invite from a friend and I haven't had the same error :(.

**Comment by andreasklinger** [comment]
Do you currently know how to reproduce this error ?

**Comment by evanp** [comment]
We've tried with several users, and Chris is the only one who sees it. And the "503" error isn't very descriptive. Is there any way to check the logs on the API?

**Comment by evanp** [comment]
Right now, it's still just @chrismessina. Chris, are you able to log into any other Product Hunt API client using "login with Product Hunt"? @andreasklinger is there a good example of doing OAuth2 authentication for PH that Chris can use to test?

**Comment by andreasklinger** [comment]
thanks for the infos

will try to take a look this week

**Comment by chrismessina** [comment]
I'm able to authenticate against this app successfully: http://www.yvoschaap.com/producthunt

**Comment by evanp** [comment]
@chrismessina They just use Twitter Auth! @andreasklinger do you have a suggestion for a site that uses the Product Hunt API OAuth flow?

**Comment by mattfogel** [comment]
@evanp I found an app that seems to be using Product Hunt OAuth: http://packhunter.co/  Does this one work for you, @chrismessina ?

**Comment by chrismessina** [comment]
Ok, I attempted to sign in to Packhunter. At first I got a "server returned no response" error, then I refreshed and I was signed in. I tried it again in an incognito window and signed in fine. 

Now I just tried personalhunt.com again in an incognito window and was able to sign in successfully...! Weird! I just tried again in my main Chrome browser identity and it worked too...! Problem solved?

**Comment by evanp** [comment]
Wuuuuuuuuuuuuuuuuut

**Comment by andreasklinger** [comment]
yay… (?)

**Comment by evanp** [comment]
Ugh. We're seeing this again. I'm trying to grab some more information on it.

**Comment by evanp** [comment]
OK, @chrismessina got the error again, and we managed to track down where it happens in the login flow.

![error_-_personal_hunt](https://cloud.githubusercontent.com/assets/6770/7703318/23962b90-fe05-11e4-9ce0-5900a7345b47.png)

We get the /me endpoint right after login to get the user's profile information. I think that's pretty standard for OAuth clients.

I've got a timestamp on the request at 2015-05-18T05:05:01.056Z . It's probably worth noting that Chris usually stress-tests social networking software by being a) an early adopter and b) having a crazy-large number of Twitter followers, which messes up everyone's follower paging code.

If I were diagnosing this problem, I'd suspect that the /me API endpoint is timing out because of Chris's large follower count.
